File: Go/company_db/backend/models/postal_code.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// PostalCode テーブル
type PostalCode struct {
	ID         int            `gorm:"primaryKey;autoIncrement" json:"id"`
	PostalCode string         `gorm:"type:varchar(10);not null;unique" json:"postal_code"`
	Address    string         `gorm:"type:varchar(255);not null" json:"address"`
	CreatedAt  time.Time      `gorm:"default:now()" json:"created_at"`
	UpdatedAt  time.Time      `gorm:"default:now()" json:"updated_at"`
	DeletedAt  gorm.DeletedAt `gorm:"index"` // ソフトデリート対応
}

File: Python/useRedis/app/models/postal_code.py
from app.db.database import Base
from sqlalchemy import TIMESTAMP, Column, Integer, String, func


class PostalCode(Base):
    __tablename__ = "postal_codes"

    id = Column(Integer, primary_key=True, index=True)
    code = Column(String, unique=True, nullable=False)
    created_at = Column(TIMESTAMP, server_default=func.now())
    updated_at = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now())
